src/main/java/com/mhackner/bamboo/AbstractGitHubStatusAction.java

```diff
@@ -60,6 +67,65 @@ void updateStatus(GHCommitState status, StageExecution stageExecution) {
         }
     }
 
+    void updateStatus(ChainExecution chainExecution) {
+        for (StageExecution stageExecution : chainExecution.getStages()) {
+            updateStatus(stageExecution);
```

Comment:
At the point this method is called, haven’t all the stages been updated? It seems like this will do it a second time.